### PR TITLE
CPP: Add detail to the CPP-205 test

### PR DESCRIPTION
**Pull request by geoffw0**
Add detail to the CPP-205 test, so that it's easier to tell which things are associated with which template / instantiation (see also the discover_walk changes https://github.com/Semmle/ql/pull/317/files#diff-9ebc04b7555032e8235243d1e995883d).

@jbj

### CODE REVIEW

**Comment by jbj** [comment]
I have no idea what this test does. Assigned @ian-semmle.

File: cpp/ql/test/library-tests/CPP-205/elements.ql [hunk 1]
```diff
@@ -1,6 +1,16 @@
 import cpp
 
+string describe(Element e) {
+  result = "from " + e.(Function).getFullSignature()
+  or
+  result = "from " + e.(FunctionDeclarationEntry).getFunction().getFullSignature()
+  or
+  result = "from " + e.(Parameter).getFunction().getFullSignature()
+  or
+  result = "from " + e.(ParameterDeclarationEntry).getFunctionDeclarationEntry().getFunction().getFullSignature()
```
Comment (ian-semmle) [inline cpp/ql/test/library-tests/CPP-205/elements.ql]: Seeing as we're enumerating these, we may as well make them even more descriptive, e.g. `"from parameter declaration entry for " + ...`

**Comment by geoffw0** [comment]
I think the test is failing because this PR assumes the presence of the `discover_walk` changes, so it should be on the `next` branch.

**Comment by ian-semmle** [comment]
The expected output now needs 4 `__va_list_tag::`s

**Comment by geoffw0** [comment]
Updated with `__va_list_tag::`.

**Comment by geoffw0** [comment]
I believe this is ready to merge.

**Pull request merged**